cupy/core/core.pyx

```diff
@@ -1974,17 +1977,11 @@ cpdef ndarray dot(ndarray a, ndarray b, ndarray out=None):
         m = 0
         n = 0
 
-    ret_shape.assign(a._shape.begin() + 1, a._shape.end())
-    ret_shape.insert(ret_shape.end(), b._shape.begin() + 1, b._shape.end())
-    if out is None:
-        if a_is_vec:
-            if b_is_vec:
-                ret_shape.clear()
-            else:
-                ret_shape.erase(ret_shape.begin())
-        elif b_is_vec:
-            ret_shape.erase(ret_shape.begin())
-    else:
+    if not a_is_vec:
```

Comment:
This `if` is unnecessary because `[a._shape.begin() + 1, a._shape.end())` is empty when `a_is_vec` is true.